async_service/trio.py

```diff
@@ -206,44 +255,20 @@ def cancel(self) -> None:
     async def wait_finished(self) -> None:
         await self._finished.wait()
 
-    def _track_current_task(
-        self, parent: trio.hazmat.Task
-    ) -> Tuple[trio.Event, trio.CancelScope]:
-        current_task = trio.hazmat.current_task()
-        if parent in self._service_task_dag:
-            self._service_task_dag[parent].append(current_task)
-
-        # We use an event to manually track when the child task is "done".
-        # This is because trio has no API for awaiting completion of a task.
-        done = trio.Event()
-        self._task_done_events[current_task] = done
-
-        # Each task gets its own `CancelScope` which is how we can manually
-        # control cancellation order of the task DAG
-        cancel_scope = trio.CancelScope()
-        self._task_cancel_scopes[current_task] = cancel_scope
-
-        # This data structure is setup so that child tasks can *inject*
-        # themselves in as dependencies.
-        self._service_task_dag[current_task] = []
```

Comment:
Some of this commentary seems worth keeping, and appears to have been dropped during the refactor.

Reply:
Good catch.  missed migrating it because it moved from function body to `__init__` code.